core/src/main/java/org/apache/calcite/interpreter/Row.java

```diff
@@ -86,6 +86,8 @@ public static Row of(@Nullable Object...values) {
   }
 
   /** Returns a copy of the values. */
+  // Note: This implements BuildInMethod.ROW_COPY_VALUES.
+  @SuppressWarnings("unused")
```

Comment:
There are lots of methods like this in the code, which are called using reflection.
I like the idea of using this annotation, but is there a guideline about which annotations should be used and which ones shouldn't?

Reply:
I'm not sure if there's a guideline for this in Calcite.
In other projects I've seen this annotation used to mark methods that are called via reflection or not used internally but needed in the public API to prevent accidental removal due to refactoring. It also has the nice side-effect of un-graying out in IDEs such as IntelliJ.